bundles/net.tourbook/src/net/tourbook/ui/UI.java

```diff
@@ -1147,6 +1267,43 @@ public static void updateUI_Tags(final TourData tourData, final Label tourTagLab
       }
    }
 
+   public static void updateUI_TagsWithImage(final Set<TourTag> tourTags, final Composite tourTagsComposite, final List<CLabel> tagsLabels) {
+
+      // We dispose the current tags labels
+      tagsLabels.forEach(tagLabel -> {
+
+         if (tagLabel != null) {
+
+            Util.disposeResource(tagLabel.getImage());
+         }
+      });
```

Comment:
Disposing these images causes...

```
org.eclipse.swt.SWTException: Graphic is disposed
	at org.eclipse.swt.SWT.error(SWT.java:4918)
	at org.eclipse.swt.SWT.error(SWT.java:4833)
	at org.eclipse.swt.SWT.error(SWT.java:4804)
	at org.eclipse.swt.graphics.Image.getBounds(Image.java:1245)
	at org.eclipse.swt.custom.CLabel.getTotalSize(CLabel.java:268)
	at org.eclipse.swt.custom.CLabel.computeSize(CLabel.java:147)
	at org.eclipse.swt.layout.RowLayout.computeSize(RowLayout.java:223)
	at org.eclipse.swt.layout.RowLayout.layoutHorizontal(RowLayout.java:291)
	at org.eclipse.swt.layout.RowLayout.computeSize(RowLayout.java:207)
	at org.eclipse.swt.widgets.Composite.computeSizeInPixels(Composite.java:217)
	at org.eclipse.swt.widgets.Control.computeSize(Control.java:663)
	at org.eclipse.swt.layout.GridData.computeSize(GridData.java:493)
	at org.eclipse.swt.layout.GridLayout.layout(GridLayout.java:225)
```

... when another tour is selected.

When I saw this code then I thought that this exception will occure and it occurred. Because the tag image from the image cache is used in serveral places, these images should not be disposed. Because they are relatively small, there should not be a problem with the image limits, they are similar like the tour type images.

Reply:
It should be fixed now. I probably forgot to update this code when I added the image cache